src/vmm/src/guest_config/aarch64/mod.rs

```diff
@@ -1,13 +1,90 @@
 // Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-use std::collections::HashMap;
+/// Module with CPU templates for aarch64
+pub mod static_cpu_templates;
 
-/// CPU configuration for aarch64 CPUs
-#[derive(Clone, Default, Debug, Eq, PartialEq)]
+use arch::regs::{Aarch64Register, Error as ArchError};
+use kvm_ioctls::VcpuFd;
+pub use static_cpu_templates::*;
+
+use super::templates::aarch64::Aarch64CpuTemplate;
+use crate::vmm_config::machine_config::CpuTemplateType;
+
+/// Errors thrown while configuring templates.
+#[derive(Debug, PartialEq, Eq, thiserror::Error)]
+#[error("Failed to create a guest cpu configuration: {0}")]
+pub struct Error(#[from] ArchError);
+
+/// CPU configuration for aarch64
+#[derive(Debug, Default, Clone, PartialEq, Eq)]
 pub struct Aarch64CpuConfiguration {
-    /// Register values as a key pair
-    /// Key: Register pointer
-    /// Value: Register value
-    pub regs: HashMap<u64, u128>,
+    /// Vector of CPU registers
+    pub regs: Vec<Aarch64Register>,
+}
+
+impl Aarch64CpuConfiguration {
+    /// Creates new guest CPU config based on the provided template
+    pub fn from_cpu_template(
+        vcpu: &VcpuFd,
+        template: &Option<CpuTemplateType>,
+    ) -> Result<Self, Error> {
+        if let Some(ref cpu_template) = template {
+            Self::with_template(vcpu, cpu_template)
+        } else {
+            Ok(Self::default())
+        }
+    }
+
+    /// Creates new guest CPU config based on the provided template
+    pub fn with_template(vcpu: &VcpuFd, template: &CpuTemplateType) -> Result<Self, Error> {
+        let cpu_config = match template {
+            CpuTemplateType::Custom(ref template) => Self::with_appied_template(vcpu, template)?,
+            CpuTemplateType::Static(template) => match template {
+                Aarch64StaticCpuTemplates::V1N1 => {
+                    let template = static_cpu_templates::v1n1::v1n1();
+                    Self::with_appied_template(vcpu, &template)?
+                }
+                _ => {
+                    unreachable!("Options other then V1N1 are invalid")
+                }
+            },
+        };
+        Ok(cpu_config)
+    }
+
+    /// Creates new guest CPU config based on the provided template
+    pub fn with_appied_template(
+        vcpu: &VcpuFd,
+        template: &Aarch64CpuTemplate,
+    ) -> Result<Self, ArchError> {
+        let mut regs = Vec::new();
+        for modifier in template.reg_modifiers.iter() {
+            let value = vcpu
+                .get_one_reg(modifier.addr)
+                .map_err(|e| ArchError::GetRegister(modifier.addr, e))?;
+            regs.push(Aarch64Register {
+                id: modifier.addr,
+                value: modifier.bitmap.apply(value),
+            });
+        }
+
+        Ok(Self { regs })
+    }
```

Comment:
You are proposing to use `TryFrom` as a constructor, but I don't really like it. Reminds me of C++ with implicit constructors. Just because methods return `Result<_,_>` does not mean we need `TryFrom`.

Added internal improvements though.

Reply:
I consider `TryFrom` the better approach as it conforms to simply naming, calling and allows for the trait on generics. But in this case I don't have a strong opinion so I am fine either way.